### PR TITLE
(maint) Additional test fixes for strict

### PR DESCRIPTION
**Pull request by tvpartytonight**

### CODE REVIEW

File: acceptance/tests/i18n/modules/puppet_agent.rb [hunk 1]
```diff
@@ -4,6 +4,9 @@
   tag 'audit:high',
       'audit:acceptance'
 
+  skip_test('i18n test module uses deprecated function; update module to resume testing.')
+  # function validate_absolute_path used https://github.com/eputnam/eputnam-i18ndemo/blob/621d06d/manifests/init.pp#L15
```
Comment (joshcooper) [inline acceptance/tests/i18n/modules/puppet_agent.rb:8]: random aside, someone else ran into a similar issue in the `puppet_agent` module https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/628

File: acceptance/tests/lookup/config5_interpolation.rb [hunk 2]
```diff
@@ -2,13 +2,16 @@
...
 
+
 tag 'audit:high',
     'audit:integration',
     'audit:refactor',  # This test specifically tests interpolation on the master.
                        # Recommend adding an additonal test that validates
                        # lookup in a masterless setup.
     'server'
 
+  pending_test('unexpected `::roles` returning undefined here, but the test passes when strict=warning')
```
Comment (joshcooper) [inline acceptance/tests/lookup/config5_interpolation.rb:13]: could you add `PUP-11751` to the message, so I remember to revisit this test

File: acceptance/tests/parser_functions/hiera/lookup_data.rb [hunk 3]
```diff
@@ -32,7 +32,6 @@
       - "yaml"
     :logger: "console"
     :hierarchy:
-      - "%{fqdn}"
```
Comment (joshcooper) [inline acceptance/tests/parser_functions/hiera/lookup_data.rb:35]: Was `fqdn` not being used in the test?

Comment (tvpartytonight) [reply acceptance/tests/parser_functions/hiera/lookup_data.rb:35]: It apparently wasn't used in any of the `lookup_data.rb` tests, but the strict erroring was causing failures.

File: acceptance/tests/lookup/config5_interpolation.rb (hunk 2 above)
Comment (tvpartytonight) [reply acceptance/tests/lookup/config5_interpolation.rb:13]: I updated the comment to specifically look at the acceptance test too.

**Review by joshcooper — APPROVED** [review]

**Pull request merged**